a10_octavia/controller/worker/tasks/vthunder_tasks.py

```diff
@@ -853,42 +853,87 @@ class TagInterfaceForMember(TagInterfaceBaseTask):
-                                                master_device_id=master_device_id)
-        except req_exceptions.ConnectionError:
-            LOG.exception("Failed to connect A10 Thunder device: %s", vthunder.ip_address)
-        except Exception as e:
-            LOG.exception("Failed to delete VLAN %s due to %s", str(vlan_id), str(e))
+        if isinstance(member, list):
+            for m in member:
+                if not m.subnet_id:
+                    LOG.warning("Subnet id argument was not specified during "
+                                "issuance of create command/API call for member %s. "
+                                "Skipping TagInterfaceForMember task", m.id)
+                    return
+                try:
+                    if vthunder and vthunder.device_network_map:
+                        vlan_id = self.get_vlan_id(m.subnet_id, False)
+                        if self.is_vlan_deletable():
+                            LOG.warning("Reverting tag interface for member "
+                                        "with VLAN id %s", vlan_id)
+                            master_device_id = vthunder.device_network_map[0].vcs_device_id
+                            for device_obj in vthunder.device_network_map:
+                                self.delete_device_vlan(vlan_id, member.subnet_id, vthunder,
+                                                        device_id=device_obj.vcs_device_id,
+                                                        master_device_id=master_device_id)
+                except req_exceptions.ConnectionError:
+                    LOG.exception("Failed to connect A10 Thunder device: %s", vthunder.ip_address)
+                except Exception as e:
+                    LOG.exception("Failed to delete VLAN %s due to %s", str(vlan_id), str(e))
+        else:
+            if not member.subnet_id:
+                LOG.warning("Subnet id argument was not specified during "
+                            "issuance of create command/API call for member %s. "
+                            "Skipping TagInterfaceForMember task", member.id)
+                return
+            try:
+                if vthunder and vthunder.device_network_map:
+                    vlan_id = self.get_vlan_id(member.subnet_id, False)
+                    if self.is_vlan_deletable():
+                        LOG.warning("Reverting tag interface for member with VLAN id %s", vlan_id)
+                        master_device_id = vthunder.device_network_map[0].vcs_device_id
+                        for device_obj in vthunder.device_network_map:
+                            self.delete_device_vlan(vlan_id, member.subnet_id, vthunder,
+                                                    device_id=device_obj.vcs_device_id,
+                                                    master_device_id=master_device_id)
+            except req_exceptions.ConnectionError:
```

Comment:
The exception handling is the same as "if isinstance(member, list)" case, so maybe we can do the try/except for both case together.

Reply:
Done